src/openrct2/title/TitleSequence.h

```diff
@@ -38,18 +38,34 @@ struct TitleCommand
     };
 };
 
-struct TitleSequence
+class TitleSequence
 {
-    std::string Path;
-    std::string Name;
+  public:
+    const std::string& Path() const { return _path; }
+    const std::string& Name() const { return _name; }
+    bool IsZip() const { return _isZip; }
 
-    std::vector<TitleCommand> Commands;
-    std::vector<std::string> Saves;
+    std::vector<TitleCommand>& Commands() { return _commands; }
+    std::vector<std::string>& Saves() { return _saves; }
 
-    size_t NumSaves() const { return Saves.size(); }
-    size_t NumCommands() const { return Commands.size(); }
+    size_t NumSaves() const { return _saves.size(); }
+    size_t NumCommands() const { return _commands.size(); }
 
-    bool IsZip;
+    void SetName(const std::string& name) { _name = name; }
+    void SetPath(const std::string& path) { _path = path; }
+    void SetIsZip(bool isZip) { _isZip = isZip; }
+
+    void AcquireCommands(std::vector<TitleCommand>& commands) { _commands = std::move(commands); }
+    void AcquireSaves(std::vector<std::string>& saves) { _saves = std::move(saves); }
+
+  private:
+    std::string _path;
+    std::string _name;
+
+    std::vector<TitleCommand> _commands;
+    std::vector<std::string> _saves;
+
+    bool _isZip {false};
```

Comment:
```suggestion
    std::string Path;
    std::string Name;

    std::vector<TitleCommand> Commands;
    std::vector<std::string> Saves;

    bool IsZip{false};
```

If you're taking the effort to make it into a class, let's also rename the parameters to be consistent with our coding style: https://github.com/OpenRCT2/OpenRCT2/wiki/Coding-Style

Reply:
I went with the CoreGuildelines and kept it as a struct:
`C.2: Use class if the class has an invariant; use struct if the data members can vary independently`

Which was the case here: (Path, Name, Saves, Commands and IsZip are all independent).